### PR TITLE
Google Analytics 4 source documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Added documentation for GA4 extract source.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/sources/extract/google-analytics-4.mdx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+---
+title: "Google Analytics 4"
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-analytics-4.mdx:2]: @am6010 in our web app, we call this Google Analytics **v4**, which can be misleading. Can we update it to Google Analytics 4 for consistency?

File: docs/sources/extract/google-analytics-4.mdx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    - **Authenticate via Google (OAuth)**: To authenticate via OAuth, click the **Sign in with Google** button, select your Google account, and give RudderStack the required permissions.
+    - **Service Account Key Authentication**: Enter your <a href="https://cloud.google.com/iam/docs/creating-managing-service-account-keys#creating">service account</a> JSON credentials in the **Service Account Information** field.
+
+<img src="../../assets/cloud-extract-sources/google-analytics-4-connection-settings-1.png" alt="Google Analytics 4 authentication" />
+
+The other settings are as follows:
+
+<img src="../../assets/cloud-extract-sources/google-analytics-4-connection-settings-2.png" alt="Google Analytics 4 dashboard settings" />
+
+- **Property ID**: Specify the <a href="https://developers.google.com/analytics/devguides/reporting/data/v1/property-id">Google Analytics 4 property identifier</a> whose events are tracked.
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-analytics-4.mdx:43]: @am6010 please confirm if this description is accurate. If no, please suggest an alternative. I referred to this description from the spec.

File: docs/sources/extract/google-analytics-4.mdx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+<img src="../../assets/cloud-extract-sources/google-analytics-4-data-import-settings.png" alt="Selecting the data to import" />
+
+<div class="infoBlock">
+Any custom reports configured using the <strong>Custom Reports</strong> dashboard setting will also be shown here.
+</div>
+
+The below table mentions the sync types supported by the Google Analytics 4 resources while syncing data to your warehouse destination:
+
+| Resource | Sync type | Primary key |
+| :---| :--- | :----|
```
Comment (ameypv-rudder) [inline docs/sources/extract/google-analytics-4.mdx:82]: @am6010 Should we add the endpoint details here as well? We are doing so for some of the other extract sources. I couldn't find the endpoint-related details in the spec.

File: docs/sources/extract/google-analytics-4.mdx (hunk 1 above)
Comment (am6010) [reply docs/sources/extract/google-analytics-4.mdx:2]: Good spot I will raise a pr to fix it.

File: docs/sources/extract/google-analytics-4.mdx (hunk 2 above)
Comment (am6010) [reply docs/sources/extract/google-analytics-4.mdx:43]: Looks good

File: docs/sources/extract/google-analytics-4.mdx (hunk 3 above)
Comment (am6010) [reply docs/sources/extract/google-analytics-4.mdx:82]: I will provide more details. Actually, it is one endpoint that processes the reports. All the preconfigured resources are reports created by default.

**Review by am6010 — COMMENTED** [review]
Looks good I will give more info about the endpoints

**Review by nidhilashkari17 — APPROVED** [review]

File: docs/sources/extract/google-analytics-4.mdx (hunk 3 above)
Comment (am6010) [reply docs/sources/extract/google-analytics-4.mdx:82]: For all reports, we use the same API. It is the `analyticsdata.googleapis.com/v1beta/` API. 
When we call this API we need to follow this convention.
```
POST https://analyticsdata.googleapis.com/v1beta/properties/GA4_PROPERTY_ID:runReport
  {
    "dateRanges": [{ "startDate": "29daysAgo", "endDate": "today" }],
    "metrics": [{ "name": "metricName" }]
    "dimensions": ["name: "dimensionName"]
  }
```

**Review by am6010 — APPROVED** [review]

File: docs/sources/extract/google-analytics-4.mdx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  color={'blueviolet'}
+/>
+
+## Getting started
+
+To set up Google Analytics 4 as a source in RudderStack, follow these steps:
+
+1. Log into your [RudderStack dashboard](https://app.rudderstack.com/).
+2. Go to **Sources** > **New source** > **Cloud Extract** and select **Google Analytics 4** from the list of sources.
+3. Assign a name to your source and click **Next**.
```
Comment (ishachopra0401) [inline docs/sources/extract/google-analytics-4.mdx:27]:
```suggestion
3. Assign a name to your source and click **Continue**.
```

File: docs/sources/extract/google-analytics-4.mdx [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+| `locations` | Incremental | `uuid` |
+| `pages` | Incremental | `uuid` |
+| `traffic_sources` | Incremental | `uuid` |
+| `website_overview` | Incremental | `uuid` |
+| `weekly_active_users` | Incremental | `uuid` |
+
+For syncing all the reports, RudderStack uses the `analyticsdata.googleapis.com/v1beta/{property_id}/runReport` API.
+
+<div class="warningBlock">
+This source only supports <strong>Incremental</strong> reports with <strong>date</strong> as dimensions.
```
Comment (ishachopra0401) [inline docs/sources/extract/google-analytics-4.mdx:95]:
```suggestion
This source supports only <strong>Incremental</strong> reports with <strong>date</strong> as dimensions.
```
> with **date** as dimensions..

Does it mean date is a required dimension and no other dimension should be added?

File: docs/sources/extract/google-analytics-4.mdx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+<img src="../../assets/cloud-extract-sources/google-analytics-4-connection-settings-1.png" alt="Google Analytics 4 authentication" />
+
+The other settings are as follows:
+
+<img src="../../assets/cloud-extract-sources/google-analytics-4-connection-settings-2.png" alt="Google Analytics 4 dashboard settings" />
+
+- **Property ID**: Specify the <a href="https://developers.google.com/analytics/devguides/reporting/data/v1/property-id">Google Analytics 4 property identifier</a> whose events are tracked.
+- **Date Range Start Date**: Choose the start date from which you want RudderStack to ingest the Google Analytics 4 data. RudderStack will not replicate any data before this date.
+- **Custom Reports**: Use this setting to sync your custom <a href="https://support.google.com/analytics/answer/10445879?hl=en">Google Analytics 4 reports</a>. You can sync multiple reports. To add a custom report, click **Add Report** and specify the **Report Name**, **Metrics**, and **Dimensions**, as shown:
```
Comment (ishachopra0401) [inline docs/sources/extract/google-analytics-4.mdx:45]: Should we also mention that a user can edit/delete a report as well?

Comment (ameypv-rudder) [reply docs/sources/extract/google-analytics-4.mdx]: Yep, definitely. Added this note.

File: docs/sources/extract/google-analytics-4.mdx (hunk 5 above)
Comment (ameypv-rudder) [reply docs/sources/extract/google-analytics-4.mdx]: You're right - `date` is a required dimension, but not the only one. Tweaked the sentence accordingly.

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**